File: java/ReadFile.java
import java.nio.file.Files;
import java.nio.file.Paths;

public class ReadFile {
    public static void main(String[] args) throws Exception {
        var filePath = Paths.get("./my-text-file.txt");

        Files.lines(filePath)
            .forEach(line -> System.out.println("line: " + line));
    }
}

File: csharp/ReadFile.cs
using System;
using System.IO;

class ReadFile
{
    public static void Main(string[] args)
    {
        var filePath = "./my-text-file.txt";

        foreach (var line in File.ReadLines(filePath))
        {
            Console.WriteLine($"line: {line}");
        }
    }
}
